generated/nodejs/test/errorreporting.js

```diff
@@ -0,0 +1,245 @@
+var USER_NAME = 'testuser';
+
+var clouderrorreportingV1beta1 =
+    require('@google-cloud/errorreporting').v1beta1();
+
+/**
+ * Integration test for ReportErrorsServiceApi.reportErrorEvent
+ *
+ * Report first two error events (to be one group) and third event
+ * (to be another group)
+ */
+function createErrorEvents() {
+  var api = clouderrorreportingV1beta1.reportErrorsServiceApi();
+  var formattedProjectName = api.projectPath(PROJECT_NAME);
+
+  var httpRequestConext = {
+    method: 'GET',
+    url: 'https://20160920t135819-dot-gapic-test.appspot.com'
+  };
+  var sourceLocation1 = {
+    file_path: 'foo/bar',
+    line_number: 20,
+    function_name: 'doGet'
+  };
+  var sourceLocation2 = {
+    file_path: 'foo/bar',
+    line_number: 30,
+    function_name: 'doSet'
+  };
+  var event1 = {
+    message: 'error reporting test event1',
+    service_context: {
+      service: SERVICE,
+      version: VERSION
+    },
+    context: {
+      http_request: httpRequestConext,
+      user: USER_NAME,
+      report_location: sourceLocation1
+    }
+  };
+  console.log('Reporting error event 1');
+  api.reportErrorEvent(formattedProjectName, event1, function(err, response) {
+    if (err) {
+      console.error(err);
+      return;
+    }
+    console.log(response);
+  });
+
+  console.log('Reporting error event 2');
+  var event2 = {
+    message: 'error reporting test event2',
+    service_context: {
+      service: SERVICE,
+      version: VERSION
+    },
+    context: {
+      http_request: httpRequestConext,
+      user: USER_NAME,
+      report_location: sourceLocation1
+    }
+  };
+  api.reportErrorEvent(formattedProjectName, event2, function(err, response) {
+    if (err) {
+      console.error(err);
+      return;
+    }
+  });
+  console.log('Reporting error event 3');
+  var event3 = {
+    message: 'error reporting test event3',
+    service_context: {
+      service: SERVICE,
+      version: VERSION
+    },
+    context: {
+      http_request: httpRequestConext,
+      user: USER_NAME,
+      report_location: sourceLocation2
+    }
+  };
+  api.reportErrorEvent(formattedProjectName, event3, function(err, response) {
+    if (err) {
+      console.error(err);
+      return;
+    }
+  });
+}
+
+/**
+ * Integration test for ErrorStatsServiceApi.listGroupStats,
+ *                      ErrorStatsServiceApi.listEvents
+ *
+ */
+function listErrorGroupStats() {
+  var api = clouderrorreportingV1beta1.errorStatsServiceApi();
+  var formattedProjectName = api.projectPath(PROJECT_NAME);
+  var timeRange = {
+    period: clouderrorreportingV1beta1.QueryTimeRange.Period.PERIOD_1_HOUR
+  };
+
+  console.log('Listing all the error groups');
+  var groups = [];
+  api.listGroupStats(formattedProjectName, timeRange).on('data',
+      function(errorGroupStat) {
+        console.log(errorGroupStat.group.name + ' ' +
+            errorGroupStat.group.group_id + ' ' +
+            errorGroupStat.count);
+        groups.push(errorGroupStat);
```

Comment:
It seems you don't have to have the full set of `groups`, right?
You can set the first state into a variable like `firstErrorGroup`, and invoke `end` to the stream so that the stream knows you don't need data anymore. That is:
```javascript
var stream = api.listGroupStats(formattedProjectName, timeRange);
stream.on('data', function(errorGroupStat) {
  console.log(...);
  // no more data.
  stream.end();
  listErrorGroupEvents(errorGroupStat.group.group_id);
});
```